pkg/fqdn/name_manager.go

```diff
@@ -139,11 +146,21 @@ func NewNameManager(config Config) *NameManager {
 		}
 	}
 
+	// Setup the update queue and a trivial goroutine to apply them
+	updateQueue := make(chan func(), 65536) // ~0.5MB of buffer
```

Comment:
I'm fine with this for addressing the backport, but for future I wonder how big this really should be.

Channel sizes should be considered the burst size of events we expect to hit. Is this GC + 65K DNS requests concurrently being handled at peak load?

Reply:
This PR, as it is now, will apply to 1.7 and master cleanly. I figured I would open a new, proper, PR on master and let the normal backports happen (the user is able to cherry pick this for now, I think).
In any case, the 65k includes DNS and GC calls (gc calls into ForceGenerateDNS). Currently, we don't see the 50ms timeout message from the DNS proxy (although I haven't been checking for it so maybe it is there). Things are much slower now so I might add this timeout message to the blacklist in tests and see how often it pops up in a few runs. If it never does, then it is more likely that the queue is not backing up that much. If it does pop up, then the queue is backing up somewhat. If we get the queue full message then I will cry since we managed to fill 65k entries. :/